addon/components/frost-chart.js

```diff
@@ -0,0 +1,294 @@
+    }
+  },
+
+  _setupProperties () {
+    this.set('_chartState.domain.x', this.get('xDomain'))
+    this.set('_chartState.domain.y', this.get('yDomain'))
+
+    const xRange = this.get('xRange')
+    if (xRange) {
+      this.set('_chartState.range.x', xRange)
+    }
+
+    const yRange = this.get('yRange')
+    if (yRange) {
+      this.set('_chartState.range.y', yRange)
+    }
+
+    this.set('_chartState.scale.x', this.get('xScale'))
+    this.set('_chartState.scale.y', this.get('yScale'))
+  },
+
+  _setupXAxis ({alignment, height, ticks, tickHeight, width}) {
+    const renderedTicks = this.get('_chartState.axes.x.renderedTicks')
+    const firstTickMargin = renderedTicks.get('firstObject.width') / 2
+    const lastTickMargin = renderedTicks.get('lastObject.width') / 2
+
+    this.get('_chartState.axes.x').setProperties({
+      alignment,
+      firstTickMargin,
+      height,
+      lastTickMargin,
+      ticks,
+      tickHeight,
+      width: width - firstTickMargin - lastTickMargin
+    })
+
+    this.incrementProperty('_chartState.axes.rendered')
+
+    const registeredAxes = this.get('_chartState.axes.registered')
+    if (registeredAxes === 0 || this.get('_chartState.axes.rendered') === registeredAxes) {
+      this._setupCanvas()
+      this.set('_chartState.axes.initialized', true)
+    }
+  },
+
+  _setupYAxis ({alignment, height, ticks, width}) {
+    const renderedTicks = this.get('_chartState.axes.y.renderedTicks')
+    const firstTickMargin = renderedTicks.get('firstObject.height') / 2
+    const lastTickMargin = renderedTicks.get('lastObject.height') / 2
+
+    this.get('_chartState.axes.y').setProperties({
+      alignment,
+      firstTickMargin,
+      height: height - firstTickMargin - lastTickMargin,
+      lastTickMargin,
+      ticks,
+      width
+    })
+
+    this.incrementProperty('_chartState.axes.rendered')
+
+    const registeredAxes = this.get('_chartState.axes.registered')
+    if (registeredAxes === 0 || this.get('_chartState.axes.rendered') === registeredAxes) {
+      this._setupCanvas()
+      this.set('_chartState.axes.initialized', true)
+    }
+  },
+
+  // == Tasks =================================================================
+
+  _resizeTask: task(function * () {
+    this._resize()
+  }).keepLatest(),
+
+  // == DOM Events ============================================================
+
+  // == Lifecycle Hooks =======================================================
+
+  didInsertElement () {
+    this._super(...arguments)
+    run.scheduleOnce('sync', this, this._setupChart)
+    if (this._hasDynamicRange()) {
+      window.addEventListener('resize', this._onResize.bind(this));
+    }
+  },
+
+  init () {
+    this._super(...arguments)
+    run.scheduleOnce('sync', this, this._setupProperties)
+  },
+
+  willDestroyElement() {
+    if (this._hasDynamicRange()) {
+      window.removeEventListener('resize', this._onResize.bind(this))
```

Comment:
This will not properly remove the resize listener, as calling bind is technically a new function.

`didInsertElement` could do something like:
```js
  this._boundedOnResize = this._onResize.bind(this)
  window.add(...)
```

`willDestroyElement`
```js
  window.removeEventListener(... this._boundedResizeListener)
```